smdebug/profiler/profiler_config_parser.py

```diff
@@ -332,19 +339,83 @@ def write_tf_dataloader_flag(self, flag_filename):
 
         return success
 
+    def _handle_step_python_profiling(self, step_phase: StepPhase, mode: ModeKeys, current_step):
+        """Handle python profiling at the given step phase, mode and step by stopping python profiling and
+        starting python profiling again if python profiling is enabled and python profiling stats should be saved
+        for the current step.
+        """
+        if not self.profiling_enabled or not self.config.python_profiling_config.is_enabled():
+            return
+
+        self.python_profiler.stop_profiling(
+            step_phase, end_mode=mode_keys_to_python_profile_mode(mode), end_step=current_step
+        )
+
+        if self.should_save_metrics(MetricsCategory.PYTHON_PROFILING, current_step):
+            self.python_profiler.start_profiling(
+                step_phase,
+                start_mode=mode_keys_to_python_profile_mode(mode),
+                start_step=current_step,
+            )
+
+    def start_pre_step_zero_python_profiling(self):
+        """Start pre-step zero python profiling if python profiling is enabled.
+        """
+        if not self.profiling_enabled or not self.config.python_profiling_config.is_enabled():
+            return
```

Comment:
nit: encapsulate this boolean with a function to reduce duplication.

Reply:
Done.